client/app/screens/CreateMatchScreen.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useEffect } from 'react';
+import { Button, ButtonText, ScrollView, FormControl, FormControlLabel, FormControlLabelText, Input, InputField, View, Text} from "@gluestack-ui/themed";
+import { AppStackParamList } from 'app/navigators/AppNavigator';
+import { NativeStackNavigationProp } from '@react-navigation/native-stack';
+import { colors, typography} from "app/theme"
+
+type CreateMatchScreenNavigationProp = NativeStackNavigationProp<
+  AppStackParamList,
+  'Welcome'
+>
+
+export interface CreateMatchScreenProps {
+  navigation: CreateMatchScreenNavigationProp;
+}
+
+/**
+ * CreateMatchScreen component allows users to create and send new match.
+ * @component
+ */
+export function CreateMatchScreen(props: CreateMatchScreenProps) {
+  const [leagueName, setLeagueName] = React.useState<string>("");
+  const [week, setWeek] = React.useState<string>("");
+  const [teamName, setTeamName] = React.useState<string>("");
+  const [createMatchDisabled, setCreateMatchDisabled] = React.useState<boolean>(true);
+  const NFL_WEEKS = 18;
+
+  // TODO: Allow matches for current/future weeks only. 
+  // Pass the current week as a prop/state to this screen.
+  // Output error message if week is invalid.
+
+  useEffect(() => {
+    if (leagueName !== "" && week !== "" && teamName !== "" && Number(week) <= NFL_WEEKS) {
```

Comment:
Two suggestions on this. The nfl week input allows for the user to enter 0 or a negative week. This really does not matter because we will be removing this input anyway. But the other thing is you can change leagueName !== "" to just leagueName because it will be read as boolean checking if the string is not empty. So I would change the if to `(leagueName && week && teamName && Number(week) <= NFL_WEEKS && Number(week) >= 1)`

Reply:
Good catch